src/core/datatypes/forte_array_common.h

```diff
@@ -29,7 +29,24 @@
  */
 template<typename T>
 class CIEC_ARRAY_COMMON : public CIEC_ARRAY {
-public:
+    friend bool operator==(const CIEC_ARRAY_COMMON &paLeft, const CIEC_ARRAY_COMMON &paRight) {
+      if (paLeft.getLowerBound() != paRight.getLowerBound() || paLeft.getUpperBound() != paRight.getUpperBound()) {
+        return false;
+      }
+
+      for (intmax_t i = paLeft.getLowerBound(), end = paRight.getUpperBound(); i <= end; ++i) {
```

Comment:
I know that the bounds of left and right are checked on equality before, but is there a special reason to use paLeft.getLowerBound() and then paRight.getUpperBound()?
If there is no reason, I would personally prefer to use bounds of either left or right, as people might wonder if there is a special reason

Reply:
Ah, makes sense. I changed it.